packages/prompts/src/index.ts

```diff
@@ -1014,3 +1014,482 @@ export const tasks = async (tasks: Task[], opts?: CommonOptions) => {
+						toggleSelected();
+						// Important: prevent the space from being added to the input
+						return false;
+					}
+					
+					// Allow typing again when user presses any other key
+					if (key !== 'up' && key !== 'down' && key !== ' ' && key !== 'return') {
+						isNavigating = false;
+					}
+					
+					// Don't block other key events
+					return;
+				});
+				
+				// Handle cursor movement
+				this.on('cursor', (key) => {
+					if (filtered.length === 0) return;
+					
+					// Enter navigation mode
+					isNavigating = true;
+					
+					if (key === 'up') {
+						cursor = cursor > 0 ? cursor - 1 : filtered.length - 1;
+					} else if (key === 'down') {
+						cursor = cursor < filtered.length - 1 ? cursor + 1 : 0;
+					}
+				});
+				
+				// Register input change handler to update filtering
+				this.on('value', () => {
+					// Only update input when not in navigation mode
+					if (!isNavigating) {
+						const newInput = this.value || '';
+						if (newInput !== input) {
+							input = newInput;
+							filterOptions(input);
+						}
+					}
+				});
+				
+				isEventsRegistered = true;
+			}
+			
+			// Handle initial filtering
+			if (this.state === 'initial') {
+				input = this.value || '';
+				filterOptions(input);
+			}
+			
+			// Handle submit state
+			if (this.state === 'submit') {
+				this.value = selectedValues as any;
+			}
+			
+			// Title and symbol
+			const title = `${color.gray(S_BAR)}\n${symbol(this.state)}  ${opts.message}\n`;
+			
+			// Selection counter
+			const counter = selectedValues.length > 0
+				? color.cyan(` (${selectedValues.length} selected)`)
+				: '';
+			
+			// Mode indicator
+			const modeIndicator = isNavigating 
+				? color.yellow(' [navigation mode]') 
+				: '';
+			
+			// Search input display
+			const searchText = isNavigating 
+				? color.dim(input) // Just show plain text when in navigation mode
+				: (this.value ? this.valueWithCursor : color.inverse(color.hidden('_')));
+				
+			const matches = filtered.length !== opts.options.length
+				? color.dim(` (${filtered.length} match${filtered.length === 1 ? '' : 'es'})`)
+				: '';
+			
+			// Render prompt state
+			switch (this.state) {
+				case 'submit': {
+					return `${title}${color.gray(S_BAR)}  ${color.dim(`${selectedValues.length} items selected`)}`;
+				}
+				case 'cancel': {
+					return `${title}${color.gray(S_BAR)}  ${color.strikethrough(color.dim(input))}`;
+				}
+				default: {
+					// Instructions
+					const instructions = isNavigating
+						? [`${color.dim('Space:')} select, ${color.dim('Enter:')} confirm, ${color.dim('Type:')} exit navigation`]
+						: [`${color.dim('Type')} to filter, ${color.dim('↑/↓')} to navigate`];
+					
+					// No results message
+					const noResults = filtered.length === 0 && input
+						? [`${color.cyan(S_BAR)}  ${color.yellow('No matches found')}`]
+						: [];
+					
+					// Get limited options for display
+					const displayOptions = limitOptions({
```

Comment:
Could we update `limitOptions` to accept formatting for the overflow? It'd be nice to indent the `...` with two spaces for this.